numba/np/arrayobj.py

```diff
@@ -1688,8 +1694,10 @@ def fancy_setslice(context, builder, sig, args, index_types, indices):
         shape_error = builder.icmp_signed('!=', index_shape[0], seq_len)
 
         with builder.if_then(shape_error, likely=False):
-            msg = "cannot assign slice from input of different size"
-            context.call_conv.return_user_exc(builder, ValueError, (msg,))
+            msg = ("cannot assign slice of shape", seq_len,
+                   "from input of shape", index_shape[0])
+            nb_types = (None, types.int32, None, types.int32)
+            context.call_conv.return_dyncamic_user_exc(builder, ValueError, msg, nb_types)
```

Comment:
There's a typo here:
```suggestion
            context.call_conv.return_dynamic_user_exc(builder, ValueError, msg, nb_types)
```